node/src/command.rs

```diff
@@ -14,10 +14,21 @@ use storage_hub_runtime::Block;
 
 use crate::{
     chain_spec,
-    cli::{Cli, RelayChainCli, Subcommand},
+    cli::{Cli, ProviderType, RelayChainCli, Subcommand},
     service::new_partial,
 };
 
+/// Configuration for the provider.
+#[derive(Debug, Clone)]
+pub struct ProviderOptions {
+    /// Provider type.
+    pub provider_type: ProviderType,
+    /// P2P port
+    pub p2p_port: u16,
+    /// Seed to generate deterministic peer id.
+    pub secret_key_seed: String,
```

Comment:
Isn't this a bad combination with `#[derive(Debug)]`? 😜

Reply:
true